### ISSUE TITLE
Wrong auto-correct on Style/SafeNavigation cop

### ISSUE THREAD

**Issue by huyvohcmc** [rubocop-hq/rubocop#5829]
```ruby
def now_what
  if user.past_organisations
    user.past_organisations.map(&:id).include?(some_id)
  end ||
  if user.current_organisations
    user.current_organisations.map(&:id).include?(some_id)
  end
end
```
Rubocop show `Style/SafeNavigation` warnings at `if user.past_organisations` and `if user.current_organisations`

## Steps to reproduce

- Run `bundle exec rubocop --auto-correct` for this file

## Expected behavior

It can fix `Style/SafeNavigation` warnings and doesn't change the logic.

## Actual behavior

This is the result:
```ruby
    def now_what
      if user.current_organisations
        user.past_organisations.map(&:id).include?(some_id) if user.past_organisations ||
        user.current_organisations.map(&:id).include?(some_id)
      end
    end
```
As you can see, the logic was changed and this is very dangerous.

## RuboCop version

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.5, running on ruby 2.3.3 x86_64-darwin17)
```

**Comment by rrosenblum** [comment]
It looks like this is an issue with the combination of `Style/IfUnlessModifier` and `Style/SafeNavigation`.

**Comment by rrosenblum** [comment]
I have never seen the usage of `||` combining `if` statements before. That seems to be the underlying problem here. 

It seems like the code could be written as
```ruby
def now_what
  user.past_organisations && user.past_organisations.map(&:id).include?(some_id) || user.current_organisations && user.current_organisations.map(&:id).include?(some_id)
end
```
Which would auto-correct to
```ruby
def now_what
  user.past_organisations&.map(&:id)&.include?(some_id) || user.current_organisations&.map(&:id)&.include?(some_id)
end
```

We will likely create a cop that recommends not using `||` or `&&` chained onto if statements.
